projects/common/src/time/page-time-range-preference.service.ts

```diff
@@ -1,53 +1,74 @@
 import { Injectable } from '@angular/core';
 import { isNil } from 'lodash-es';
-import { Observable } from 'rxjs';
+import { combineLatest, Observable } from 'rxjs';
 import { map, shareReplay, take } from 'rxjs/operators';
+import { ApplicationFeature } from '../constants/application-constants';
+import { FeatureStateResolver } from '../feature/state/feature-state.resolver';
+import { FeatureState } from '../feature/state/feature.state';
 import { NavigationService } from '../navigation/navigation.service';
 import { PreferenceService, StorageType } from '../preference/preference.service';
+import { RelativeTimeRange } from './relative-time-range';
+import { TimeDuration } from './time-duration';
 import { TimeRange } from './time-range';
 import { TimeRangeService } from './time-range.service';
+import { TimeUnit } from './time-unit.type';
 
 @Injectable({ providedIn: 'root' })
 export class PageTimeRangePreferenceService {
   private static readonly STORAGE_TYPE: StorageType = StorageType.Local;
   private static readonly TIME_RANGE_PREFERENCE_KEY: string = 'page-time-range';
 
+  private readonly globalDefaultTimeRange: TimeRange = new RelativeTimeRange(new TimeDuration(30, TimeUnit.Minute));
+
   private readonly pageTimeRangeStringDictionary$: Observable<PageTimeRangeStringDictionary>;
 
   public constructor(
     private readonly preferenceService: PreferenceService,
     private readonly timeRangeService: TimeRangeService,
-    private readonly navigationService: NavigationService
+    private readonly navigationService: NavigationService,
+    private readonly featureStateResolver: FeatureStateResolver
   ) {
     this.pageTimeRangeStringDictionary$ = this.buildPageTimeRangeObservable();
   }
 
-  public getTimeRangePreferenceForPage(path: string): Observable<TimeRange | undefined> {
-    return this.pageTimeRangeStringDictionary$.pipe(
-      map(pageTimeRangeStringDictionary => {
-        if (isNil(pageTimeRangeStringDictionary[path])) {
-          return this.getDefaultPageTimeRange(path);
+  public getTimeRangePreferenceForPage(rootLevelPath: string): Observable<TimeRange> {
+    return combineLatest([
+      this.pageTimeRangeStringDictionary$,
+      this.featureStateResolver.getCombinedFeatureState([
+        ApplicationFeature.PageTimeRange,
+        ApplicationFeature.NavigationRedesign
```

Comment:
> The NavigationRedesign feature enables the time range to behave with page level functionality

Oh interesting, so this is not what I would expect from the naming. I would have expected the behavior changes (the time range is scoped to the page) to come from the PageTimeRange flag, as well as the relocation of the control at the same time - basically all the work in this PR. I thought the Navigation redesign was dependent on that (since the new structure has no time range selector), but otherwise unrelated.

So first off, if we really want that separation, it seems like there's a flag rename in order (but let's see).

> We could just use NavigationRedesign in this instance, however in the case that we disabled page time range and enable NavigationRedesign the ui will be broken because

We shouldn't try to add complexity to allow any matrix of flags to work. We can assume dependencies if it saves us significant work, I think that was mentioned earlier. In that scenario you described though, how are we solving the lack of TR? If we've removed top and disabled page, it's just not a valid state. And In that same scenario, given the time range page behavior is part of the Nav redesign flag as you described, we would actually want 2.